naga-cli/src/bin/naga.rs

```diff
@@ -505,15 +559,32 @@ fn parse_input(
     input: Vec<u8>,
     params: &Parameters,
 ) -> Result<Parsed, Box<dyn std::error::Error>> {
-    let (module, input_text) = match Path::new(&input_path)
-        .extension()
-        .ok_or(CliError::from("Input filename has no extension"))?
-        .to_str()
-        .ok_or(CliError::from("Input filename not valid unicode"))?
-    {
-        "bin" => (bincode::deserialize(&input)?, None),
-        "spv" => naga::front::spv::parse_u8_slice(&input, &params.spv_in).map(|m| (m, None))?,
-        "wgsl" => {
+    let input_kind = match params.input_kind {
+        Some(kind) => kind,
+        None => {
+            let ext = input_path
+                .extension()
+                .ok_or(CliError::from("Input filename has no extension"))?
+                .to_str()
+                .ok_or(CliError::from("Input filename not valid unicode"))?;
+            InputKind::from_str(ext).or_else(|_| {
+                InputKind::from_str(
+                    Path::new(ext)
+                        .extension()
```

Comment:
This second call of `extension()` is unnecessary because it actually `extension()` always returns the last part of the extension e.g. `.glsl` for `.frag.glsl`.  Fixed in [7ac8e39](https://github.com/gfx-rs/wgpu/pull/5411/commits/7ac8e39955695ed9ad1ee39105e0268e3b665625)